urlb44e9843ba_QuasimodoNZ_SWEN222_Game_Project_tgz-pJ8-Game_World_GameWorldLevelBuilderJ8/src/Graphics_Renderer/GraphicsRenderer.java

```diff
@@ -0,0 +1,320 @@
+	 * Tileset, it will make an image and draw each tile on and return the final
+	 * product
+	 * 
+	 * @param tileIDs
+	 * @param tileset
+	 * @return
+	 */
+	public static Image combineTiles(int[][] tileIDs, Tileset tileset) {
+		//Sets up the image that will be returned
+		BufferedImage image = new BufferedImage(tileIDs[0].length
+				* tileset.getTileWidth(), tileIDs.length
+				* tileset.getTileHeight(), BufferedImage.TYPE_4BYTE_ABGR);
+		//Seperates out a graphics from the image so that the image can be manipulated
+		Graphics canvas = image.getGraphics();
+		
+		//Draws each tile at its corresponding x and y coordinates based on the size of each tile
+		for (int x = 0; x < tileIDs[0].length; x++)
+			for (int y = 0; y < tileIDs.length; y++)
+				if (tileIDs[y][x] != -1)
+					canvas.drawImage(tileset.getTile(tileIDs[y][x]), x
+							* tileset.getTileWidth(),
+							y * tileset.getTileHeight(), null);
+
+		return image;
+	}
+
+	/**
+	 * Given a Graphics and Container, will draw a grid for the Container and
+	 * draw all the icon versions of the objects onto the grid
+	 * 
+	 * @param g
+	 * @param c
+	 */
+	public static void drawInventory(Graphics g, Container c) {
+		
+		//Draws tiles for each available slot in the container
+		int size = c.getInventorySize();
+		if (size != 4 && size != 9 && size != 16)
+			throw new IllegalArgumentException(
+					"The container c must have a size of 4, 9 or 16. Its size is "
+							+ size);
+		for (int x = 0; x < Math.sqrt(size); x++)
+			for (int y = 0; y < Math.sqrt(size); y++)
+				try {
+					g.drawImage(ImageIO.read(new File(
+							"Images/Inventory Square.png")), x * 38,
+							y * 38, null);
+				} catch (IOException e) {
+					e.printStackTrace();
+				}
+
+		//Draws each item in the container in its allocated tile slot
+		for (int i = 0; i < c.getItems().size(); i++) {
+			int x = (int) (i % Math.sqrt(size)) * 38 + 2;
+			int y = (int) (i / Math.sqrt(size)) * 38 + 2;
+			g.drawImage(c.getItems().get(i).toIcon(), x, y, null);
+		}
+	}
+
+	/**
+	 * If the static map allGraphics contains the key string, it returns the
+	 * image otherwise it initialises the image and returns it
+	 * 
+	 * @param key
+	 * @return
+	 */
+	public static Image getImage(String key) {
+		if (allGraphics.containsKey(key))
+			return allGraphics.get(key);
+		else
+			return initialiseImage(key);
+	}
+
+	/**
+	 * If the static map allTilesets contains the key string, it returns the
+	 * tileset otherwise it initialises the tileset and returns it
+	 * 
+	 * @param key
+	 * @return
+	 */
+	public static Tileset getTileset(String key) {
+		if (allTilesets.containsKey(key))
+			return allTilesets.get(key);
+		else
+			return initialiseTileset(key);
+	}
+
+	/**
+	 * It creates an Image based on the string key, it then adds it to the
+	 * static map allGraphics and then returns the image
+	 * 
+	 * @param key
+	 * @return
+	 */
+	private static Image initialiseImage(String key) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 244-258 into a separate method.